### ISSUE TITLE
Undefined reference to __stack_chk_fail on Linux

### ISSUE THREAD

**Issue by johnor** [evilcorpltd/stdlib2#13]
gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)

test_types.c:(.text+0x69): undefined reference to `__stack_chk_fail'
collect2: error: ld returned 1 exit status
test/CMakeFiles/types.dir/build.make:84: recipe for target 'test/types' failed
make[2]: *** [test/types] Error 1
CMakeFiles/Makefile2:128: recipe for target 'test/CMakeFiles/types.dir/all' failed
make[1]: *** [test/CMakeFiles/types.dir/all] Error 2
Makefile:94: recipe for target 'all' failed

**Comment by johnor** [comment]
Looks like it is related to -fstack-protector

man gcc says:

           NOTE: In Ubuntu 14.10 and later versions, -fstack-protector-strong is enabled by default for
           C, C++, ObjC, ObjC++, if none of -fno-stack-protector, -nostdlib, nor -ffreestanding are
           found.

Not sure why it is still used although we have -nostdlib though

**Comment by robinlinden** [comment]
Yeah, I couldn't disable run-time checks in debug builds on Windows either. Hence the sad regex hack to remove the flags:
```
STRING(REGEX REPLACE "[/|-]RTC(su|[1su])" "" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
```

**Comment by robinlinden** [comment]
Resolved by #15
